### PR TITLE
added controllers, mappers, dto requests and dto responses also added…

### PR DESCRIPTION
**Pull request by Alekearl**
… new methods in dao layer and service

### CODE REVIEW

File: src/main/java/com/dev/cinema/controller/CinemaHallController.java [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+@RequestMapping("/cinema-halls")
+public class CinemaHallController {
+    private CinemaHallService cinemaHallService;
+    private CinemaHallMapper cinemaHallMapper;
```
Comment (engineer027) [inline src/main/java/com/dev/cinema/controller/CinemaHallController.java]:
```suggestion
    private final CinemaHallService cinemaHallService;
    private final CinemaHallMapper cinemaHallMapper;
```

File: src/main/java/com/dev/cinema/controller/MovieController.java [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+@RequestMapping("/movies")
+public class MovieController {
+    private MovieService movieService;
+    private MovieMapper movieMapper;
```
Comment (engineer027) [inline src/main/java/com/dev/cinema/controller/MovieController.java]: теж саме

File: src/main/java/com/dev/cinema/controller/MovieSessionController.java [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+@RequestMapping("/movie-sessions")
+public class MovieSessionController {
+    private MovieSessionMapper movieSessionMapper;
+    private MovieSessionService movieSessionService;
```
Comment (engineer027) [inline src/main/java/com/dev/cinema/controller/MovieSessionController.java]: теж саме

File: src/main/java/com/dev/cinema/model/dto/MovieDtoRequest.java [hunk 4]
```diff
@@ -0,0 +1,31 @@
+package com.dev.cinema.model.dto;
+
+public class MovieDtoRequest {
+    private Long id;
+    private String title;
+    private String description;
```
Comment (engineer027) [inline src/main/java/com/dev/cinema/model/dto/MovieDtoRequest.java]:
```suggestion
    private String title;
    private String description;
```

Comment (engineer027) [reply src/main/java/com/dev/cinema/model/dto/MovieDtoRequest.java]: ми в реквест не сетимо id, нам БД призначає

File: src/main/java/com/dev/cinema/service/mapper/MovieSessionMapper.java [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+import com.dev.cinema.service.MovieService;
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+@Component
+public class MovieSessionMapper {
+    private CinemaHallService cinemaHallService;
+    private MovieService movieService;
```
Comment (engineer027) [inline src/main/java/com/dev/cinema/service/mapper/MovieSessionMapper.java]: final

File: src/main/java/com/dev/cinema/controller/MovieSessionController.java (hunk 3 above)
Comment (Alekearl) [reply src/main/java/com/dev/cinema/controller/MovieSessionController.java]: Ok

File: src/main/java/com/dev/cinema/service/mapper/MovieSessionMapper.java (hunk 5 above)
Comment (Alekearl) [reply src/main/java/com/dev/cinema/service/mapper/MovieSessionMapper.java]: Ok.

File: src/main/java/com/dev/cinema/controller/MovieController.java (hunk 2 above)
Comment (Alekearl) [reply src/main/java/com/dev/cinema/controller/MovieController.java]: ok

**Review by engineer027 — APPROVED** [review]

File: src/main/java/com/dev/cinema/model/dto/MovieSessionDtoRequest.java [hunk 6]
```diff
@@ -0,0 +1,40 @@
+package com.dev.cinema.model.dto;
+
+public class MovieSessionDtoRequest {
+    private Long movieId;
+    private String movieTitle;
```
Comment (KseniiaMakarova) [inline src/main/java/com/dev/cinema/model/dto/MovieSessionDtoRequest.java]: do we need this field? is it really informative for creating MovieSession object?

File: src/main/java/com/dev/cinema/model/dto/MovieSessionDtoResponse.java [hunk 7]
```diff
@@ -0,0 +1,49 @@
+package com.dev.cinema.model.dto;
+
+public class MovieSessionDtoResponse {
+    private Long id;
+    private Long movieId;
```
Comment (KseniiaMakarova) [inline src/main/java/com/dev/cinema/model/dto/MovieSessionDtoResponse.java:5]: is this field set in your mapper?

**Review by KseniiaMakarova — CHANGES_REQUESTED** [review]

File: src/main/java/com/dev/cinema/model/dto/MovieSessionDtoResponse.java (hunk 7 above)
Comment (Alekearl) [reply src/main/java/com/dev/cinema/model/dto/MovieSessionDtoResponse.java:5]: Fixed.

File: src/main/java/com/dev/cinema/model/dto/MovieSessionDtoRequest.java (hunk 6 above)
Comment (Alekearl) [reply src/main/java/com/dev/cinema/model/dto/MovieSessionDtoRequest.java]: I'm pretty sure that we need Id's. But If say for real, I can't answer you for sure do we need this field or not and will it be informative. So I'll just remove it. O__o

**Review by KseniiaMakarova — APPROVED** [review]

File: src/main/java/com/dev/cinema/model/dto/MovieSessionDtoRequest.java (hunk 6 above)
Comment (KseniiaMakarova) [reply src/main/java/com/dev/cinema/model/dto/MovieSessionDtoRequest.java]: we need ids in this case, and we can get the whole entity by id, so there is no need to pass the title as well

**Pull request merged**